### ISSUE TITLE
Frozen dataclass error

### ISSUE THREAD

**Issue by Chrischu2503** [hgrecco/pint#2073]
In `pint\delegates\txt_defparser\common.py", line 23`, it states `@dataclass(frozen=True)`.
Apparently, that clashes with 
`Python313\Lib\dataclasses.py", line 1295, in wrap 
                          return _process_class(cls, init, repr, eq, order, unsafe_hash,
                          frozen, match_args, kw_only, slots,
                          weakref_slot)`
due to `Python313\Lib\dataclasses.py", line 1043, in _process_class
    raise TypeError('cannot inherit frozen dataclass from a '
                    'non-frozen one')
TypeError: cannot inherit frozen dataclass from a non-frozen one`

My local hard fix was to comment in the line 23.
I am runing Windows 11, installed python 3.13.0 via the official webside and am using VScode.

**Comment by dopplershift** [comment]
Duplicate of #1969.

**Comment by Zippy1999** [comment]
I am seeing the same problem but on python `3.11.9` , only in the past 24 hours. 

```
<module>
    @dataclass(frozen=True)
../../../../.pyenv/versions/3.11.9/lib/python3.11/dataclasses.py:1222: in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
../../../../.pyenv/versions/3.11.9/lib/python3.11/dataclasses.py:993: in _process_class
    raise TypeError('cannot inherit frozen dataclass from a '
E   TypeError: cannot inherit frozen dataclass from a non-frozen one
```

Everything was working yesterday but after reinstalling the project's virtual env / deps this morning,  I am seeing this error

**Comment by hgrecco** [comment]
I am closing this issue as is fixed in Pint 0.24.4 Please let me know if you encounter any problem.
